src/AutomatedCar/SystemComponents/Sensors/Sensor.cs

```diff
@@ -31,13 +32,65 @@ public Sensor(VirtualFunctionBus virtualFunctionBus)
             this.triangle = new SensorTriangle();
         }
 
+        protected abstract List<WorldObject> FilterRelevantWorldObjects();
+
+//        protected abstract void SaveWorldObjectsToPacket(List<WorldObject> worldObjects);
         protected abstract void SaveWorldObjectsToPacket();
 
-        protected abstract List<WorldObject> FilterRelevantWorldObjects();
+        protected AutomatedCar GetAutomatedCar()
+        {
+            return World.Instance.ControlledCar;
+        }
 
         protected List<WorldObject> GetWorldObjects()
         {
             return World.Instance.WorldObjects;
         }
+
+        /// <summary>
+        /// Converts degree to radian.
+        /// </summary>
+        /// <param name="angleDeg">Angle in degree.</param>
+        /// <returns>Angle in radian.</returns>
+        protected double AngleDegToRad(double angleDeg)
+        {
+            return Math.PI * angleDeg / 180;
+        }
+
+        /// <summary>
+        /// Rotates a 2D coordinate with angle given in degree around (0,0).
+        /// </summary>
+        /// <param name="pos">Position in 2D.</param>
+        /// <param name="angleDeg">Angle given in degree.</param>
+        /// <returns>Rotated coordinate.</returns>
+        protected Vector2 RotatePosition(Vector2 pos, double angleDeg)
+        {
+            double angleRad = this.AngleDegToRad(angleDeg);
+            return new Vector2(
+                (int)((pos.X * Math.Cos(angleRad)) - (pos.Y * Math.Sin(angleRad))),
+                (int)((pos.X * Math.Sin(angleRad)) + (pos.Y * Math.Cos(angleRad))));
+        }
+
+        /// <summary>
+        /// Returns whether point is in a triangle with Barycentric method.
+        /// </summary>
+        /// <param name="p">Point</param>
+        /// <param name="p0">Triangle Point 1.</param>
+        /// <param name="p1">Triangle Point 2.</param>
+        /// <param name="p2">Triangle Point 3.</param>
+        /// <returns>Point is in triangle or not.</returns>
+        protected bool PointInTriangle(Vector2 p, Vector2 p0, Vector2 p1, Vector2 p2)
```

Comment:
Ez a fv. lesz CollisionDetectionben. Emiatt blokkolva vagy annak mergeléséig. 

Amúgy szép, átlátható kód!